### PR TITLE
Fix for 3 issues

### PR DESCRIPTION
**Pull request by Lucksmith369**
Issue: #2277 #2235 #2214

### 1. Why is this change necessary?

Fix 3 Issues and make PWA usable for Products without image or Manufacturer

### 2. What does this change do, exactly?

Fix a bug if $parentId is null

### 3. Describe each step to reproduce the issue or behaviour.

As Example, create Shopware PWA and try to access ProductDetail Page where Product doesnt have a Picture

### 4. Please link to the relevant issues (if any).

#2277 #2235 #2214

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Comment by shyim** [comment]
Can you add a failing test?

**Comment by Lucksmith369** [comment]
> Can you add a failing test?

Can do tommorow dont have more time today.

**Comment by shyim** [comment]
It's ok. I will mark it as incomplete until then

File: src/Core/Framework/DataAbstractionLayer/Dbal/EntityReader.php [hunk 1]
```diff
@@ -546,7 +546,11 @@ private function loadOneToManyWithPagination(
             $parentId = $entity->get('parentId');
 
             //extract mapping ids for the current entity
+            $mappingIds = [];
+			
+			if($parentId){
```
Comment (King-of-Babylon) [inline src/Core/Framework/DataAbstractionLayer/Dbal/EntityReader.php:551]: If no parentId is found you could easily continue the foreach.
Please change it that way.

And like @shyim wrote: Add some tests for this case :blush:

**Comment by keulinho** [comment]
Also please sign the Contributor License Agreement, otherwise we sadly can not merge your PR.

**Comment by Lucksmith369** [comment]
> Also please sign the Contributor License Agreement, otherwise we sadly can not merge your PR.

cant sign its open a page with noch CLA in it

CLA-assistant

for Testcase you can simply call an api with association

**Comment by Lucksmith369** [comment]
> > Also please sign the Contributor License Agreement, otherwise we sadly can not merge your PR.
> 
> cant sign its open a page with noch CLA in it
> 
> CLA-assistant
> 
> for Testcase you can simply call an api with association

ok after 10 times retry signing was possible.

**Comment by shyim** [comment]
Hey,

I will close this as it's inactive for a while. Feel free to create a new PR when it's finished

**Pull request closed without merging**